contribute/architecture/state.md

```diff
@@ -0,0 +1,142 @@
+# State
+
+Grafana uses a database to persist settings between restarts. In fact, if you don't specify one, Grafana creates a [SQLite3](https://www.sqlite.org/) database file on your local disk. This guide explains how to store and retrieve data from the database.
+
+Grafana supports the following databases:
+
+- [MySQL](https://www.mysql.com/)
+- [PostgreSQL](https://www.postgresql.org/)
+- [SQLite3](https://www.sqlite.org/)
+
+Grafana uses the [XORM](https://xorm.io) framework for persisting objects to the database. For more information on how to use XORM, refer to the [documentation](http://gobook.io/read/github.com/go-xorm/manual-en-US/).
+
+[Services](services.md) don't use XORM directly. Instead, services use the _SQL store_, a special type of service that provides an abstraction for the database layer. There are two ways of using the `sqlstore`: using `sqlstore` handlers, and using the `SqlStore` instance.
+
+## `sqlstore` handlers
+
+> **Deprecated:** We are deprecating `sqlstore` handlers in favor of using the `SqlStore`  object directly in each service. Since most services still use the `sqlstore` handlers, we still want to explain how they work.
```

Comment:
```suggestion
> **Deprecated:** We are deprecating `sqlstore` handlers in favor of using the `SqlStore` object directly in each service. Since most services still use the `sqlstore` handlers, we still want to explain how they work.
```

Reply:
This is an extra space between `SqlStore` and object but GitHub is being dumb.